dependencies.yaml

```diff
@@ -346,15 +346,13 @@ dependencies:
         packages:
           - pip
           - pip:
-              # TODO: Figure out what to do with this dependency
-              # since the repo is now archived.
-              - git+https://github.com/dask/dask-glm@main
+              - dask-glm==0.3.0
           # TODO: remove pin once a release that includes fixes for the error
           # is released: https://github.com/rapidsai/cuml/issues/5514
           - hdbscan<=0.8.30
       - output_types: pyproject
         packages:
-          - dask-glm @ git+https://github.com/dask/dask-glm@main
+          - dask-glm==0.3.0
             # TODO: Can we stop pulling from the master branch now that there was a release in October?
```

Comment:
I think comment technically belongs to the dask-glm, but I'm hesitant to delay merge of this.